pymedphys/_dicom/anonymise.py

```diff
@@ -466,10 +467,13 @@ def is_anonymised_dataset(ds, ignore_private_tags=False):
                     float(elem.value), float(dummy_value)
                 ):
                     continue
-
+                logging.info("%s is not considered to be anonymised", elem.name)
```

Comment:
See the following table:


Level | When it’s used
-- | --
DEBUG | Detailed information, typically of interest only when diagnosing problems.
INFO | Confirmation that things are working as expected.
WARNING | An indication that something unexpected happened, or indicative of some problem in the near future (e.g. ‘disk space low’). The software is still working as expected.
ERROR | Due to a more serious problem, the software has not been able to perform some function.
CRITICAL | A serious error, indicating that the program itself may be unable to continue running.

from -- https://docs.python.org/3/howto/logging.html#when-to-use-logging

Based on that table, is info the right logging level here?

Reply:
I believe so.  Things are working as expected for the programmer.
It isn't the library programmer who is debugging whether their data is anonymised or not.
It is the user or API programmer who wants to understand why their data is not considered anonymised.
is_anonymised_dataset is a public function.  If it were private (e.g. _is_anonymised_dataset() and only intended to be called by testing code, or was itself test code only), I'd still consider it to be info level if one expected the return value to be true in the vast majority of the typical use cases.

There are a number of different arguments to the CLI that would result in different behaviour operating on the same data.  So the use of varying arguments might result in outcomes that the user might need to understand, but the library programmer would consider to be absolutely correct behaviour that does not require debugging.  The library programmer would think that there might be a user training issue (assuming The Fine Manual was clear, and simply had not been Read...)

Consider what you would do if someone said "this isn't working right, my data is being called not anonymised, but I anonymised it according to the same rules as pymedphys as far as I can tell".  Yes, you could tell them to put it in debug logging and then slog through the debug logs.  Or you could ask them to log at the info level to confirm that things are operating as expected.

The distinction I make is whether the "problems" are unexpected or not for considering it DEBUG.
DEBUG content is only beautiful to the programmer who put the content in.  Stuff is not working as expected so the programmer needs to know what happened before, what happened after, and needs significant context to understand what might be going on.
INFO content should be meaningful to a service engineer or support person, or knowledgable user.

The fact that I ran in to this while acting as a library programmer and found a defect in the code is separate from the need for a user to be able to receive confirmation that things (a dataset is found not to be anonymised) are working as expected (by being informed of the reason why the dataset is not considered to be anonymised).